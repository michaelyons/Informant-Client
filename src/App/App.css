text {
  cursor: pointer;
}

text:hover {
  font-weight: 900;
}

div .hero-body {
  padding-top: 0.5rem;
}

.modal-content {
  color: black;
  background-color: rgba(0, 0, 0, 0.01);
}

.card-header-title {
  background-color: rgba(5, 0, 0, 0.2);
}

.notification {
  padding-left: 0;
}

.john-head {
  margin-bottom: 0;
}

.reset-btn-home {
  background-color: #151515;
  border: none;
  color: white;
  padding: 7px 30px;
  margin-left: 0.5rem;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin-bottom: 1.5rem;
  cursor: pointer;
}

.input-home-field {
  border: 1px solid #979a97;
  background-color: #e9e9e9;
  padding-left: 0.4rem;
  width: 30%;
  height: 37.5px;
  font-size: 16px;
  margin-bottom: 1.5rem;
}

.reset-btn-cloud {
  background-color: #151515;
  border: none;
  color: white;
  padding: 7px 30px;
  margin-left: 0.5rem;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin-bottom: 1rem;
  cursor: pointer;
}

.wordcloud-input {
  border: 1px solid #979a97;
  background-color: #e9e9e9;
  padding-left: 0.4rem;
  width: 30%;
  height: 37.5px;
  font-size: 16px;
  margin-bottom: 1rem;
}

iframe {
  margin-top: 1.5rem;
}

.lobbyist-name {
  margin-right: 1rem;
}
