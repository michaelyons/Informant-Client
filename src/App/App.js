import React, { Component } from 'react';
import CardContainer from '../CardContainer/CardContainer';
import { Route, Switch } from 'react-router-dom';
import Navigation from '../Navigation/Navigation';
import {
  initialFetchCall,
  lobbyistFetchCall,
  wordCloudFetch
} from '../Utils/apiCalls';
import WordCloud from 'react-d3-cloud';

import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      recentTopics: [],
      lobbyistList: [],
      wordCloud: [],
      errors: ''
    };
  }

  componentDidMount() {
    this.setInitialState();
    this.fetchLobbyists();
    this.setWordCloud();
  }

  setInitialState = async () => {
    if (!this.state.recentTopics.length) {
      try {
        const recentTopics = await initialFetchCall();
        this.setState({ recentTopics });
      } catch (error) {
        this.setState({ errors: error.message });
      }
    }
  };

  setWordCloud = async () => {
    if (!this.state.wordCloud.length) {
      try {
        const wordCloud = await wordCloudFetch();
        this.setState({ wordCloud });
      } catch (error) {
        this.setState({ errors: error.message });
      }
    }
  };

  fetchLobbyists = async () => {
    if (!this.state.lobbyistList.length) {
      try {
        const lobbyists = await lobbyistFetchCall();
        this.setState({ lobbyistList: lobbyists });
      } catch (error) {
        this.setState({ error: error.message });
      }
    }
  };

  render() {
    const fontSizeMapper = word => Math.log2(word.value) * 3;
    return (
      <div className="app">
        <Navigation />
        <WordCloud
          data={this.state.wordCloud}
          fontSizeMapper={fontSizeMapper}
        />
        <main>
          <div>
            <Switch>
              <Route
                path="/"
                render={() => {
                  return (
                    <CardContainer recentTopics={this.state.recentTopics} />
                  );
                }}
              />
              <Route
                path="/lobbyists"
                render={() => {
                  this.fetchLobbyists();
                  return (
                    <CardContainer lobbyistList={this.state.lobbyistList} />
                  );
                }}
              />
            </Switch>
          </div>
        </main>
      </div>
    );
  }
}

export default App;
