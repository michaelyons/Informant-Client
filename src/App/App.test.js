import React from 'react';
import App from './App';
import { shallow } from 'enzyme';
import { mockRecentTopicsFetch } from '../mockData';

describe('App Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
    // window.fetch = jest.fn().mockImplementation(() =>
    //   Promise.resolve({
    //     json: () => Promise.resolve(mockDataFetch)
    //   })
    // );
  });

  describe('snapshot', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
