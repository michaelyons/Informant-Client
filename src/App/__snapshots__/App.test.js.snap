// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component snapshot should match the snapshot 1`] = `undefined`;
